KPI/H.adoc

```diff
@@ -1,235 +1,239 @@
-            </gmd:MD_Keywords>
- <gmd:descriptiveKeywords>
+```
 
+.Keywords implementation rules
+|===
+|Rule |Score
 
-....
+a|`gmd:keywords` are user-defined. There is no presence of `gmd:type` or
+`gmd:thesaurusName`
+|1
 
-and
+|`gmd:keywords`, `gmd:type`, and `gmd:thesaurusName` are defined. Terms are
+defined by `gco:CharacterString` elements, not URIs via `gmx:Anchor`
+|1
 
-<! -- check for well referenced thesaurus using the Anchor with a resolvable HTTP URI -- >
+|Compliance rules are implemented and they conform to well defined keywords rules.
+|2
 
- //gmd:MD_DataIdentification/gmd:descriptiveKeywords/gmd:MD_Keywords/gmd:thesaurusName/gmd:CI_Citation/gmd:title/gmx:Anchor/@xlink:href
+|Codelist and values are checked against the references provided and HTTP URIs
+can be successfully resolved
+|2
+|===
 
 
-e.g.
-....
-<gmd:descriptiveKeywords>
-            <gmd:MD_Keywords>
-               <gmd:thesaurusName>
- 		<gmd:CI_Citation>
-		 <gmd:title>
-		  <gmx:Anchor xlink:href="http://codes.wmo.int/grib2/codeflag/_4.2">WMO Codes Registry GRIB edition 2</gmx:Anchor>
-		 </gmd:title>
-		 <gmd:date>
- 		  <gmd:CI_Date>
-		   <gmd:date>
- 		    <gco:Date>2016-06-25</gco:Date>
-		   </gmd:date>
-		   <gmd:dateType>
-			 <gmd:CI_DateTypeCode codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#CI_DateTypeCode" codeListValue="publication">publication</gmd:CI_DateTypeCode>
-                   </gmd:dateType>
-   		  </gmd:CI_Date>
-		 </gmd:date>
- 		</gmd:CI_Citation>
-	       </gmd:thesaurusName>
-            </gmd:MD_Keywords>
- <gmd:descriptiveKeywords>
-....
-
-or
+=== Score values
 
-<!-- Check referenced thesaurus via CharacterStrings  --> 
+Absence of Keywords should return a non compliance result. All WMCP documents
+must provide at least one keyword from the `WMO_CategoryCode` codelist
+(checked in KPI A).
 
-<!-- use URL within CharacterString is not recommended, Anchor should be used instead-->
+=== Score values
 
- //gmd:MD_DataIdentification/gmd:descriptiveKeywords/gmd:MD_Keywords/gmd:thesaurusName/gmd:CI_Citation/gmd:title/gco:CharacterString
+.Good quality title score values
+|===
+|Rank | Score
 
-e.g.
+|Low
+|0
 
-....
-<gmd:descriptiveKeywords>
-            <gmd:MD_Keywords>
-               <gmd:thesaurusName>
- 		<gmd:CI_Citation>
-		 <gmd:title>
- 		  <gco:CharacterString>WMO Codes Registry - GRIB edition 2</gco:CharacterString>
-		 </gmd:title>
-		 <gmd:date>
- 		  <gmd:CI_Date>
-		   <gmd:date>
- 		    <gco:Date>2016-06-25</gco:Date>
-		   </gmd:date>
-		   <gmd:dateType>
-			 <gmd:CI_DateTypeCode codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#CI_DateTypeCode" codeListValue="publication">publication</gmd:CI_DateTypeCode>
-                   </gmd:dateType>
-   		  </gmd:CI_Date>
-		 </gmd:date>
- 		</gmd:CI_Citation>
-	       </gmd:thesaurusName>
-            </gmd:MD_Keywords>
- <gmd:descriptiveKeywords>
-....
+|Medium
+|1 -3
 
-=== H.4 Score Values
+|High
+|4 - 6 
```

Comment:
so the score is 0-6 for each kpi ? 
6 is the maximum score?

Reply:
current 6 is the maximum score, but open to suggestions.

Reply:
I was wonder what's the rationale behind the numbers (I think @gaubert presented that each KPI would score independently )  Final report from KPI is a set of numbers from each KPI??

Reply:
Wouldn't each KPI have a possible score, and then the entire KPI suite is presented as a total, and/or be sent to the scoring rubric to classify accordingly (score of "a", "b", "c", weighted scores, etc.) as far as I understand #53.

Reply:
each possible score (the maximum value that each KPI can score individually ) can be different ? so far we have 
B  max score 5
H max score 6
L max score 3
J max score 10 
That already puts some implicit weight on the individual KPI's , right ?

Reply:
yes, but the scoring rubric should weigh thereafter accordingly?

Another example if the NASA OpenSearch validator: https://opensearch-ui.earthdata.nasa.gov/validations/index.  Every test suite has a different number of tests, and each suite evaluates to a percentage.

It sounds like we need to harmonize the scoring rules. I will open another issue.